File: ts/2048/2016.ts
function maximumDifference(nums: number[]): number {
  const l = nums.length
  let ans = -1, min = nums[0]
  for (let i = 1; i < l; i++) {
    if (nums[i] > min) {
      ans = Math.max(ans, nums[i] - min)
    } else {
      min = nums[i]
    }
  }
  return ans
};

File: c/2048/2016.c
int maximumDifference(int* nums, int numsSize){
  int min = nums[0], ans = -1;
  for (int i = 1; i < numsSize; i++) {
    int diff = nums[i] - min;
    if (diff > 0) {
      ans = diff > ans ? diff : ans;
    } else {
      min = nums[i];
    }
  }
  return ans;
}